### PR TITLE
gwyddion: support darwin

### PR DESCRIPTION
**Pull request by cgevans**
###### Motivation for this change
Gwyddion will build on Darwin, but gnome2.gtkglext is broken there,
and so I am disabling it as an optional dependency here when building
on Darwin.

###### Things done

Note that I personally have only tested that this will continue to build and work on NixOS.  Someone else tested that the previous Gwyddion derivation builds, all binaries work, and the Python 2.7 module loads, so long as gnome2.gtkglext is not used (it fails to build on OS X); that is what this change is meant to ensure.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/chemistry/gwyddion/default.nix [hunk 1]
```diff
@@ -45,7 +45,7 @@ stdenv.mkDerivation rec {
   
   buildInputs = with stdenv.lib;
     [ gtk2 fftw ] ++
-    optionals openglSupport [gnome2.gtkglext] ++
+    optional (openglSupport && (!stdenv.hostPlatform.isDarwin)) gnome2.gtkglext ++
```
Comment (veprbl) [inline pkgs/applications/science/chemistry/gwyddion/default.nix:48]: Why not instead change `openglSupport ? true` to `openglSupport ? !stdenv.isDarwin`.

Comment (cgevans) [reply pkgs/applications/science/chemistry/gwyddion/default.nix]: Because I hadn't thought of that... thank you.

**Comment by veprbl** [comment]
@GrahamcOfBorg eval
@GrahamcOfBorg build gwyddion

**Comment by cgevans** [comment]
It looks like the libunique derivation does not have darwin set as supported, though it will build if allowUnsupportedSystem is set, and that's causing the ofborg build to fail.

File: pkgs/applications/science/chemistry/gwyddion/default.nix [hunk 2]
```diff
@@ -10,7 +10,7 @@
   libxsltSupport ? true, libxslt ? null,
   fitsSupport ? true, cfitsio ? null,
   zlibSupport ? true, zlib ? null,
-  libuniqueSupport ? true, libunique ? null,
+  libuniqueSupport ? !stdenv.isDarwin, libunique ? null,
```
Comment (veprbl) [inline pkgs/applications/science/chemistry/gwyddion/default.nix:13]: If it builds on darwin, let's just add darwin to `libunique.meta.platforms`.

Comment (cgevans) [reply pkgs/applications/science/chemistry/gwyddion/default.nix]: I was a bit concerned about doing that without testing it myself first on OS X; I've just been able to check this for both libunique and gwyddion, and so added a commit to do this.  I'm not sure if I should create a separate pull request for libunique, however?

Comment (veprbl) [reply pkgs/applications/science/chemistry/gwyddion/default.nix]: These are small changes, so they are fine to do in one PR.

**Review by veprbl — APPROVED** [review]
Passes nixpkgs-review on x86_64-darwin:
```
3 package built:
gwyddion libunique python27Packages.gwyddion
```

**Comment by veprbl** [comment]
Thank you!

**Pull request merged**